### PR TITLE
Misc Cppcheck fixes

### LINKED ISSUE #22702: Cppcheck errors in 3.1
Cppcheck version 1.76.1
Godot version 3.1 rev caa141a1ac630144e441f89689baa1a74bb3bae5

I fixed the easy ones in the PR below.

All remaining cppcheck errors in non third party code:

[core/io/logger.cpp:196]: (error) va_list 'list_copy' used before va_start() was called.
[core/io/logger.cpp:200]: (error) va_list 'list_copy' used before va_start() was called.
[core/io/logger.cpp:202]: (error) va_list 'list_copy' used before va_start() was called.
[core/io/logger.cpp:250]: (error) va_list 'list_copy' used before va_start() was called.
[core/io/logger.cpp:251]: (error) va_list 'list_copy' used before va_start() was called.
[core/io/logger.cpp:252]: (error) va_list 'list_copy' used before va_start() was called.
[core/os/memory.cpp:101]: (error) Memory leak: mem
[editor/progress_dialog.cpp:60]: (error) Uninitialized struct member: t.vb
[editor/progress_dialog.cpp:60]: (error) Uninitialized struct member: t.state
[editor/progress_dialog.cpp:121]: (error) Uninitialized variable: no_updates
[editor/progress_dialog.cpp:183]: (error) Uninitialized struct member: t.hb
[main/input_default.cpp:407]: (error) Uninitialized struct member: action.strength
[main/input_default.cpp:541]: (error) Uninitialized struct member: action.strength
[main/input_default.cpp:552]: (error) Uninitialized struct member: action.strength
[platform/iphone/os_iphone.cpp:566]: (error) Memory leak: new_ptr
[servers/physics/body_pair_sw.cpp:130]: (error) Uninitialized struct member: contact.local_shape
[servers/physics/body_pair_sw.cpp:136]: (error) Uninitialized struct member: contact.local_shape
[servers/physics/body_pair_sw.cpp:130]: (error) Uninitialized struct member: contact.collider_shape
[servers/physics/body_pair_sw.cpp:136]: (error) Uninitialized struct member: contact.collider_shape
[servers/physics/body_pair_sw.cpp:130]: (error) Uninitialized struct member: contact.active
[servers/physics/body_pair_sw.cpp:136]: (error) Uninitialized struct member: contact.active
[servers/visual/shader_language.cpp:2527]: (error) Uninitialized struct member: pos.char_idx
[servers/visual/shader_language.cpp:2527]: (error) Uninitialized struct member: pos.tk_line

Probably false positives:
[drivers/gles2/shader_gles2.h:240]: (error) Array 'values[3]' accessed at index 3, which is out of bounds.
[drivers/gles2/shader_gles2.h:256]: (error) Array 'values[3]' accessed at index 3, which is out of bounds.
[drivers/gles2/shader_gles2.h:272]: (error) Array 'values[3]' accessed at index 3, which is out of bounds.

### PR DESCRIPTION
**Pull request by swenner**
Partially fixes #22702.

### CODE REVIEW

File: editor/plugins/canvas_item_editor_plugin.h [hunk 1]
```diff
@@ -280,6 +280,8 @@ class CanvasItemEditor : public VBoxContainer {
 		Transform2D xform;
 		float length;
 		uint64_t last_pass;
+
+		BoneList() : length(0.f), last_pass(0) {}
```
Comment (LikeLakers2) [inline editor/plugins/canvas_item_editor_plugin.h]: https://travis-ci.org/godotengine/godot/jobs/436896280#L537

File: modules/gdnative/nativescript/api_generator.cpp [hunk 2]
```diff
@@ -292,6 +292,7 @@ List<ClassAPI> generate_c_api_classes() {
 				method_api.has_varargs = method_bind && method_bind->is_vararg();
 
 				// Method flags
+				method_api.is_virtual = false;
```
Comment (LikeLakers2) [inline modules/gdnative/nativescript/api_generator.cpp]: I'm not sure I understand the need for this line. Doesn't `method_api.is_virtual` get set later in the file?

File: editor/plugins/canvas_item_editor_plugin.h (hunk 1 above)
Comment (swenner) [reply editor/plugins/canvas_item_editor_plugin.h]: It's a default constructor to fix the following cppcheck errors. I initialized length as a free bonus.

[editor/plugins/canvas_item_editor_plugin.cpp:3067]: (error) Uninitialized struct member: b.last_pass
[editor/plugins/canvas_item_editor_plugin.cpp:3081]: (error) Uninitialized struct member: b.last_pass
[editor/plugins/canvas_item_editor_plugin.cpp:3097]: (error) Uninitialized struct member: b.last_pass

I will reformat it to please clang.

File: modules/gdnative/nativescript/api_generator.cpp (hunk 2 above)
Comment (swenner) [reply modules/gdnative/nativescript/api_generator.cpp]: [modules/gdnative/nativescript/api_generator.cpp:305]: (error) Uninitialized struct member: method_api.is_virtual

if method_info.flags on line 296 is false we will reassign an undefined value of method_api.is_virtual to method_api.is_virtual on line 306.

Comment (LikeLakers2) [reply modules/gdnative/nativescript/api_generator.cpp]: Nevermind, I forgot C++ isn't guaranteed to give bools a default `false` value. Woops.

You can mark this as resolved.

File: core/image.cpp [hunk 3]
```diff
@@ -780,8 +780,8 @@ void Image::resize(int p_width, int p_height, Interpolation p_interpolation) {
 
 	// Setup mipmap-aware scaling
 	Image dst2;
-	int mip1;
-	int mip2;
+	int mip1 = 0;
+	int mip2 = 0;
 	float mip1_weight;
```
Comment (akien-mga) [inline core/image.cpp]: Might be worth initializing this one too for consistency.

Comment (akien-mga) [reply core/image.cpp]: Actually I fixed that one which was also a compiler warning.

**Comment by akien-mga** [comment]
Thanks for the fixes, I've rebased them and merged as 10718b096.

**Pull request closed without merging**